api/filtering/db_filters.py

```diff
@@ -153,8 +153,11 @@ def per_reporter_staleness_filter(staleness, reporter, host_type_filter):
     return staleness_conditions
 
 
-def _staleness_filter(staleness: List[str], host_type_filter: Set[str]) -> List:
-    staleness_obj = serialize_staleness_to_dict(get_staleness_obj())
+def _staleness_filter(staleness: List[str], host_type_filter: Set[str], identity=None) -> List:
+    if identity:
+        staleness_obj = serialize_staleness_to_dict(get_staleness_obj(identity))
+    else:
+        staleness_obj = serialize_staleness_to_dict(get_staleness_obj())
```

Comment:
`get_staleness_obj()` already has handling for `identity=None`, and that's its default value, so:
```suggestion
    staleness_obj = serialize_staleness_to_dict(get_staleness_obj(identity))
```